### PR TITLE
user registration

### PR DESCRIPTION
**Pull request by Jestermaxrko**

### CODE REVIEW

File: controllers/user-controller.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    const user = await User.findOne({ email: req.body.email });
+    if (user) return res.status(409).json({ message: 'User is already exists' });
+    return next();
+  } catch (err) { return next(err) }
+}
+
+userController.validate = () => (
+  [
+    body('firstname', 'Firstnme doesnt exists').exists()
+      .isLength({ min: 2 }).withMessage('Firstname must be at least 6 character long')
```
Comment (mkhodan) [inline controllers/user-controller.js]: 6 characters? What about Oleg, Taras, etc. ?

File: services/utils.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
+const bcrypt = require('bcrypt');
+
+module.exports.hashData = async data => {
+  const salt = await bcrypt.genSalt(10);
+  return await bcrypt.hash(data, salt);
+}
```
Comment (mkhodan) [inline services/utils.js]: Please, add blank line in the end of a file.

**Review by mkhodan — CHANGES_REQUESTED** [review]

File: controllers/user-controller.js (hunk 1 above)
Comment (Jestermaxrko) [reply controllers/user-controller.js]: fixed

File: services/utils.js (hunk 2 above)
Comment (Jestermaxrko) [reply services/utils.js]: fixed

**Pull request merged**